integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/columnLineage/customCollectors/UnionFieldDependencyCollector.java

```diff
@@ -0,0 +1,47 @@
+package io.openlineage.spark3.agent.lifecycle.plan.columnLineage.customCollectors;
+
+import static io.openlineage.spark3.agent.lifecycle.plan.columnLineage.FieldDependenciesCollector.traverseExpression;
+
+import io.openlineage.spark.agent.util.ScalaConversionUtils;
+import io.openlineage.spark3.agent.lifecycle.plan.columnLineage.ColumnLevelLineageBuilder;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.IntStream;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.spark.sql.catalyst.expressions.Attribute;
+import org.apache.spark.sql.catalyst.expressions.ExprId;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.catalyst.plans.logical.Union;
+
+@Slf4j
+public class UnionFieldDependencyCollector implements CustomFieldDependencyCollector {
```

Comment:
Out of scope of this PR, but would be great to have documentation about 

- what particular LogicalPlan node is 
- from which SQL expression it's created